src/app/documentation-messages.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { NotificationService, NotificationType } from 'patternfly-ng';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DocumentationMessagesService {
+
+  private messages: Map<string, string>;
+  private readonly ready = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    private http: HttpClient,
+    private notifications: NotificationService,
+  ) {
+    this.setLocale(navigator.language);
+  }
+
+  setLocale(locale: string = navigator.language) {
```

Comment:
Hmm, setting a default value on `locale` seems funny. Why would a consumer of this API need to call `setLocale()` with no args? Is that intended as a "reset" call?

Reply:
Ah, when writing javascript, I have a (potentially bad) habit of setting default values that make little sense. Thank you for pointing out.

In the future, if we ever need the ability to reset to browser's default language, a dedicated `reset()` function should be introduced. Or just explicitly call `.setLocale(navigator.langauge)`. :)